File: 201-300/leetcode_235.c
#include <stdio.h>
#include <stdlib.h>
struct TreeNode {
    int val;
    struct TreeNode *left;
    struct TreeNode *right;
};
int main()
{
    return 0;
}
bool re(struct TreeNode* root, struct TreeNode* p);
struct TreeNode* lowestCommonAncestor(struct TreeNode* root, struct TreeNode* p, struct TreeNode* q) {
    if(root == NULL)
        return false;
    struct TreeNode* a = root;
    while(re(a,p)&&re(a,q))
    {
        if(re(a->right, p)&&re(a->right, q))
            a = a->right;
        else if(re(a->left, p)&&re(a->left, q))
            a = a->left;
        else
            break;
    }
    return a;
}

bool re(struct TreeNode* root, struct TreeNode* p){
    if(root == NULL)
        return false;
    else if(root == p)
        return true;
    else
        return re(root->left, p)||re(root->right, p);
}

File: 201-300/leetcode_235.py
class Solution:
    def lowestCommonAncestor(self, root, m1, m2):
        if not root:
            return None
        p, q = min([m1.val, m2.val]), max([m1.val, m2.val])
        x = root
        while True:
            if p <= x.val and q >= x.val:
                return x
            elif x.val < p:
                x = x.right
            else:
                x = x.left
